### DISCUSSION TITLE
Raspberry Pi or a Virtual Machine?

### DISCUSSION THREAD

**Discussion by AbdulraheemSB** [AdguardTeam/AdGuardHome#7051] [Q&A]
Greetings everyone!, I have no experience whatsoever on how this type of adblocks work. This is my first time I hear about it. I saw some YT videos to get educated on this matter.

Some video tutorials install Adguard on a Virtual Machine and some on a Raspberry Pi. 

Before I spend money on Raspberry Pi, could someone please explain what is the difference between using it over a Virtual Machine?.

Thank you.

**Comment by rafalohaki** [comment]
there can be virtual machine running on ur pc, so ur adguard dns server will run only when u have virtual machine running.
there is virtual machine server which can run 24/7 like a raspberrypi so there is not much diff, except virtual resources given by raspberpi or other hardware. for example there can be 2 virtual machine server on 1 raspberrypi but in the end its all about how much resources we get.
sipmly put. if you meant virtual machine server, then to choose between those two, you need to have in mind, which one is cheaper for long term, do you have external ip on house, so you could use dns server outside of house, etc.

**Reply by AbdulraheemSB** [reply]
Hey thanks for the input!

So if I want to keep the Adguard running all the time then Raspberry Pi would be the pick because a VM on my PC means that the PC has to be running all the time. Correct?

**Reply by rafalohaki** [reply]
yes

**Reply by AbdulraheemSB** [reply]
Thank you very much. 

One last thing if you don't mind. This just came across my mind. Is it possible if I used one of the AWS virtual machines and installed Adguard there and made the VM run the entire time?  I believe it will be costly but is it possible right?

**Reply by rafalohaki** [reply]
yes, there is like 12months of free aws virtual machine offer. you can install system there via panel and you have running virtual machine ready to install adguard home.

**Reply by AbdulraheemSB** [reply]
That's nice.

Thanks for your help & time.
